src/fetch-wrapper.ts

```diff
@@ -32,109 +14,46 @@ export default function fetchWrapper(
-    headers: Object.fromEntries(
-      Object.entries(requestOptions.headers).map(([name, value]) => [
-        name,
-        String(value),
-      ]),
-    ),
-    signal: requestOptions.request?.signal,
-    // duplex must be set if request.body is ReadableStream or Async Iterables.
-    // See https://fetch.spec.whatwg.org/#dom-requestinit-duplex.
-    ...(requestOptions.body && { duplex: "half" }),
-  })
-    .then(async (response) => {
-      url = response.url;
-      status = response.status;
-
-      for (const keyAndValue of response.headers) {
-        headers[keyAndValue[0]] = keyAndValue[1];
-      }
-
-      if ("deprecation" in headers) {
-        const matches =
-          headers.link && headers.link.match(/<([^>]+)>; rel="deprecation"/);
-        const deprecationLink = matches && matches.pop();
-        log.warn(
-          `[@octokit/request] "${requestOptions.method} ${
-            requestOptions.url
-          }" is deprecated. It is scheduled to be removed on ${headers.sunset}${
-            deprecationLink ? `. See ${deprecationLink}` : ""
-          }`,
-        );
-      }
-
-      if (status === 204 || status === 205) {
-        return;
-      }
-
-      // GitHub API returns 200 for HEAD requests
-      if (requestOptions.method === "HEAD") {
-        if (status < 400) {
-          return;
-        }
-
-        throw new RequestError(response.statusText, status, {
-          response: {
-            url,
-            status,
-            headers,
-            data: undefined,
-          },
-          request: requestOptions,
-        });
-      }
-
-      if (status === 304) {
-        throw new RequestError("Not modified", status, {
-          response: {
-            url,
-            status,
-            headers,
-            data: await getResponseData(response),
-          },
-          request: requestOptions,
-        });
-      }
-
-      if (status >= 400) {
-        const data = await getResponseData(response);
-
-        const error = new RequestError(toErrorMessage(data), status, {
-          response: {
-            url,
-            status,
-            headers,
-            data,
-          },
-          request: requestOptions,
-        });
+  const log = requestOptions.request?.log || console;
+  const parseSuccessResponseBody =
+    requestOptions.request?.parseSuccessResponseBody !== false;
 
+  const body =
+    isPlainObject(requestOptions.body) || Array.isArray(requestOptions.body)
+      ? JSON.stringify(requestOptions.body)
+      : requestOptions.body;
+
+  const requestHeaders = Object.fromEntries(
+    Object.entries(requestOptions.headers).map(([name, value]) => [
+      name,
+      String(value),
+    ]),
+  );
+
+  let fetchResponse: Response;
+
+  try {
+    fetchResponse = await fetch(requestOptions.url, {
+      method: requestOptions.method,
+      body,
+      redirect: requestOptions.request?.redirect,
+      // Header values must be `string`
```

Comment:
Can you move this comment up to the definition of `requestHeaders`

Reply:
Sure.